[Test] org.eclipse.ui.tests.navigator.resources.NestedResourcesTests.testProjectHierarchy fails on linux intermittently

The test  org.eclipse.ui.tests.navigator.resources.NestedResourcesTests.testProjectHierarchy failed on linux in the I-Build [I20220706-1800](https://download.eclipse.org/eclipse/downloads/drops4/I20220706-1800/testresults/html/org.eclipse.ui.tests.navigator_ep425I-unit-cen64-gtk3-java11_linux.gtk.x86_64_11.html) with the below error

```
expected:<2> but was:<1>

java.lang.AssertionError: expected:<2> but was:<1>
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.junit.Assert.assertEquals(Assert.java:633)
at org.eclipse.ui.tests.navigator.resources.NestedResourcesTests.testProjectHierarchy(NestedResourcesTests.java:98) 
```

--- a/tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/resources/NestedResourcesTests.java
+++ b/tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/resources/NestedResourcesTests.java
@@ -81,19 +81,29 @@ public class NestedResourcesTests {
 		testProjects.add(projectABA);
 		testProjects.add(projectABB);
 
+		// Wait for NestedProjectManager to process all resource changes
+		assertTrue("NestedProjectManager did not update children for projectA",
+				DisplayHelper.waitForCondition(Display.getDefault(), TIMEOUT,
+						() -> NestedProjectManager.getInstance().getDirectChildrenProjects(projectA).length == 1));
+
 		IProject[] childrenOfProjectA = NestedProjectManager.getInstance().getDirectChildrenProjects(projectA);
-		Assert.assertEquals(1, childrenOfProjectA.length);
 		Assert.assertEquals(projectAB, childrenOfProjectA[0]);
 		Assert.assertNull(NestedProjectManager.getInstance().getMostDirectOpenContainer(projectA));
 
+		// Wait for NestedProjectManager to process projectAAA
+		assertTrue("NestedProjectManager did not update children for folderAA",
+				DisplayHelper.waitForCondition(Display.getDefault(), TIMEOUT,
+						() -> NestedProjectManager.getInstance().getDirectChildrenProjects(folderAA).length == 1));
+
 		IProject[] childrenOfFolderAA = NestedProjectManager.getInstance().getDirectChildrenProjects(folderAA);
-		Assert.assertEquals(1, childrenOfFolderAA.length);
 		Assert.assertEquals("aaa", childrenOfFolderAA[0].getName());
 		Assert.assertEquals(folderAA,
 				NestedProjectManager.getInstance().getMostDirectOpenContainer(childrenOfFolderAA[0]));
 
-		IProject[] childrenOfProjectAB = NestedProjectManager.getInstance().getDirectChildrenProjects(projectAB);
-		Assert.assertEquals(2, childrenOfProjectAB.length);
+		// Wait for NestedProjectManager to process both projectABA and projectABB
+		assertTrue("NestedProjectManager did not update children for projectAB",
+				DisplayHelper.waitForCondition(Display.getDefault(), TIMEOUT,
+						() -> NestedProjectManager.getInstance().getDirectChildrenProjects(projectAB).length == 2));
 	}
 
 	@Test

